website_product_comment_purchased/models/mail.py

```diff
@@ -22,37 +22,38 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from openerp.osv import osv, fields
+from openerp import models, fields
 
 
-class MailMessage(osv.Model):
+class MailMessage(models.Model):
     _inherit = 'mail.message'
 
     def _comment_bought(self, cr, uid, ids, name, arg, context=None):
-        res = {}
+        res = {}.fromkeys(ids, False)
         sale_report_obj = self.pool.get('sale.report')
-        product_obj = self.pool.get('product.product')
-        cr.execute("""
-            {0}
-            FROM ( {1} )
-            {2}
-            """.format(sale_report_obj._select(), sale_report_obj._from(),
-                       sale_report_obj._group_by()))
         mail_cache = self.browse(cr, uid, ids, context)[0]
```

Comment:
You can avoid get a browse of all ids if you will get just the first item with:
`mail_cache = self.browse(cr, uid, ids[0], context)`